xblock/fields.py

```diff
@@ -498,11 +498,15 @@ def __set__(self, xblock, value):
         Setting a value does not update the underlying data store; the
         new value is kept in the cache and the xblock is marked as
         dirty until `save` is explicitly called.
+
+        Note, if there's already a cached value and it's equal to the value
+        we're trying to cache, we won't do anything.
         """
         value = self._check_or_enforce_type(value)
-        # Mark the field as dirty and update the cache:
-        self._mark_dirty(xblock, EXPLICITLY_SET)
-        self._set_cached_value(xblock, value)
+        if self._get_cached_value(xblock) != value:
```

Comment:
@cpennington , in the tests, I've run into scenarios where this line fails because the value of the field stored on the xblock can't be compared to the value we're trying to set on it. For example, if `self._get_cached_value(xblock)` is a non-timezone-aware `datetime` object, and `value` _is_ a timezone-aware `datetime` object, then trying to compare them throws a `TypeError`. Is this an error condition I should be anticipating and catching? I assume we only have timezone-aware datetimes in production, but it's possible in other scenarios we try to set a field to something it can't currently be compared to.